lib/intacct/base.rb

```diff
@@ -61,9 +65,17 @@ def persisted?
     def format_date(date_object)
       return if date_object.blank?
 
+      date_object = parse_date(date_object)
+
       if date_object.is_a?(Date) || date_object.is_a?(Time)
         date_object.strftime(DATE_FORMAT)
-      elsif date_object.is_a?(String)
+      else
+        date_object
+      end
+    end
+
+    def parse_date(date_object)
```

Comment:
Just curious, what is the benefit of breaking this out into a separate method? Seems like the functionality is essentially the same but now split across two methods.

Reply:
One formats it parses a string into a date. Since legacy endpoint have the structure of
```
<detafield>
  <year>2020</year>
  <month>9</month>
  <day>24</day>
</datefield>
```
Whereas other endpoints only seem to support `09/24/2020`. Since we don't don't the input this allows us to only parse a possible string , while the above method also formats it back to a string.

Reply:
Maybe I'm a bit confused, but couldn't `date_object.is_a?(String)` just be another case within `format_date` (how it is before the changes)?

Reply:
`format_date` returns a `String`. `parse_date` returns a `Date`.

Reply:
On L79 `strftime` will return a string.

Reply:
Right. It's not supposed to. Glad you caught that.